grakn-graql/src/main/java/ai/grakn/graql/internal/query/predicate/ComparatorPredicate.java

```diff
@@ -62,6 +65,13 @@
                 value = ((Integer) value).longValue();
             }
 
+            this.originalValue = Optional.of(value);
+
+            // Convert dates to epoch time to reflect how they are stored
+            if (value instanceof LocalDateTime) {
+                value = ResourceType.DataType.DATE.getPersistenceValue((LocalDateTime) value);
```

Comment:
Yeah! Oh, just realised: I should do this for _all_ values, not just `LocalDateTime` right?